cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -318,26 +320,26 @@ public void recordProgramStop(ProgramId programId, String pid, long stopTs, Prog
     write(key, new RunRecordMeta(started, stopTs, runStatus));
   }
 
-  public List<RunRecordMeta> getRuns(ProgramRunStatus status, Predicate<RunRecordMeta> filter) {
-    return getRuns(null, status, Long.MIN_VALUE, Long.MAX_VALUE, Integer.MAX_VALUE, filter);
+  public Map<ProgramRunId, RunRecordMeta> getRuns(ProgramRunStatus status, Predicate<RunRecordMeta> filter) {
+    return getRuns(null, status, 0L, Long.MAX_VALUE, Integer.MAX_VALUE, filter);
   }
 
-  public List<RunRecordMeta> getRuns(@Nullable ProgramId programId, ProgramRunStatus status,
+  public  Map<ProgramRunId, RunRecordMeta> getRuns(@Nullable ProgramId programId, ProgramRunStatus status,
                                      long startTime, long endTime, int limit,
                                      @Nullable Predicate<RunRecordMeta> filter) {
     if (status.equals(ProgramRunStatus.ALL)) {
-      List<RunRecordMeta> resultRecords = Lists.newArrayList();
-      resultRecords.addAll(getActiveRuns(programId, startTime, endTime, limit, filter));
-      resultRecords.addAll(getSuspendedRuns(programId, startTime, endTime, limit - resultRecords.size(), filter));
-      resultRecords.addAll(getHistoricalRuns(programId, status, startTime, endTime,
-                                             limit - resultRecords.size(), filter));
-      return resultRecords;
+      Map<ProgramRunId, RunRecordMeta> resultmap =  new HashMap<>();
```

Comment:
Hi Terence, 
Wouldn't the map keep the same ordering as the list being currently used.  
Thanks
Deepak